webrtc.html

```diff
@@ -1196,19 +1206,23 @@ <h4>Set the RTCSessionDescription</h4>
             with a newly [= exception/create | created =]
             {{InvalidStateError}} and abort these steps.</p>
           </li>
+          <li>
+            <p>Let <var>jsepSetOfTransceivers</var> be a copy of
+            <var>connection</var>'s [=set of transceivers =].</p>
+          </li>
           <li>
             <p data-tests="">In parallel, start the process to apply <var>description</var>
-              as described in <span data-jsep=
-                                  "processing-a-local-desc processing-a-remote-desc">[[!JSEP]]</span>,
-              with the additional restriction that if applying <var>description</var>
-              leads to modifying a transceiver <var>transceiver</var>, and
-              <var>transceiver</var>.<a>[[\Sender]]</a>.<a>[[\SendEncodings]]</a>
-              is non-empty, and not equal to the encodings that would result from
-              processing <var>description</var>, the process of applying <var>description</var>
-              fails.
-              This specification does not allow remotely initiated RID renegotiation.
-            </p>
-
+            as described in <span data-jsep=
+            "processing-a-local-desc processing-a-remote-desc">[[!JSEP]]</span>,
+            with these additional restrictions: Use <var>jsepSetOfTransceivers</var>
```

Comment:
This may or may not do what we want, but I'd like more input from the working group on how to read the webrtc-pc versus JSEP specs. Because one way to resolve things is to say "copy things" and another is to add a note and say that the set of objects **are in fact different**, and clarify places that are needed when implications are not clear (like with addTransceiver and addTrack).

The reason I care is there may be edge-case differences, so we need to be careful here.

For example, if we are copying the states here, that means that we may be copying states of transceivers that have not yet been synchronized with state changes from the JSEP layer of a previous SLD or SRD operation. The JS state is NOT GUARANTEED TO BE UP-TO-DATE.

Example:
1. Starting point: our transceiver is not stopped
2. Apply SRD that stops the transceiver. This is successful and a task is posted to tell the JS layer of this.
3. Without awaiting (thus the PostTask has not happened yet): Perform another SRD. Because transceiver states are copied, we perform the second SRD as if the transceiver was not stopped. PostTask to tell JS layer of the result.
4. First SRD's PostTask synchronizes JS transceivers with result of SRD operation. Transceiver is stopped.
5. Second SRD's PostTask synchronizes JS transcievers with result of SRD operation. Transceiver is not stopped.

We just un-stopped a transceiver... what?

JS cannot be an authority on the true state of the underlying system. JS can only be a control knob that acts at a distance to inform JSEP what to do. But since JSEP can do things in parallel, JSEP is the authority and JS is the thing syncing up with JSEP, in my view.

Reply:
The example works with swapping out mids too.

transciever.mid == null
SRD("mid:0")
SRD("mid:banana")

Is this now valid? When second SRD is applied mid is still null according to the copy, so assigning it "banana" should be fine, yes?

Reply:
Or are we protected from this because PostTask must have happened before the second call, because without it the promise is not resolved, and without the promise resolving the operation chain protects us? :) I hope so

Reply:
I think this is not an issue because of the operations chain, but keeping this thread unresolved for the sake of discussion later today

Reply:
I agree the operations chain protects us from this happening.